avocado/plugins/runners/exec_test.py

```diff
@@ -122,6 +122,9 @@ def _get_env(self, runnable):
         env = dict(os.environ)
         if runnable.kwargs:
             env.update(runnable.kwargs)
+            for key, value in runnable.kwargs.items():
+                if value is None:
+                    del env[key]
```

Comment:
I mentioned this before, but I think it got lost with @clebergnu 's resolved comment.  I believe you need a check for `key in env`, other wise this `del` will generate a `KeyError` if the key wasn't in the original environment.

Reply:
I see your point, but IIUIC the check is not needed, because of line 124. `env.update(runnable.kwargs)`. After this update, the `kwargs` keys become a subset of `env` keys, therefore the key will be always in `env` and we won't get `KeyError`.